### ISSUE TITLE
Alignment error with malloc call in C header

### ISSUE THREAD

**Issue by ffreling** [ziglang/zig#11402]
### Zig Version

0.10.0-dev.1736+93e11b824

### Steps to Reproduce

There are C libraries that are shipped as single-header, with an implementation flag (usually LIB_IMPLEMENTATION).

If we have a Foo library, we can define it in a single file:
```c
#pragma once

int foo();

#ifdef FOO_IMPLEMENTATION
#include <stdlib.h>

int foo() {
  char **b = malloc(sizeof(*b));
  return 0;
}

#endif

```

Then we can import it in zig:
```zig
const std = @import("std");
const foo = @cImport({
    @cDefine("FOO_IMPLEMENTATION", {});
    @cInclude("foo.h");
});

pub fn main() anyerror!void {
    const f = foo.foo();
    std.log.info("{} your codebase are belong to us.", .{f});
}
```

### Expected Behavior

Importing a single-header C library should works just like in C.

### Actual Behavior

Using the single header to compile the library leads to this error:
```
./zig-cache/o/c0006cc78d7a58721f2bef21f13c4dbe/cimport.zig:476:25: error: cast increases pointer alignment
    var b: [*c][*c]u8 = @ptrCast([*c][*c]u8, @alignCast(@import("std").meta.alignment([*c]u8), malloc(@sizeOf([*c]u8))));
                        ^
./zig-cache/o/c0006cc78d7a58721f2bef21f13c4dbe/cimport.zig:476:46: note: '?*anyopaque' has alignment 1
    var b: [*c][*c]u8 = @ptrCast([*c][*c]u8, @alignCast(@import("std").meta.alignment([*c]u8), malloc(@sizeOf([*c]u8))));
                                             ^
./zig-cache/o/c0006cc78d7a58721f2bef21f13c4dbe/cimport.zig:476:34: note: '[*c][*c]u8' has alignment 8
    var b: [*c][*c]u8 = @ptrCast([*c][*c]u8, @alignCast(@import("std").meta.alignment([*c]u8), malloc(@sizeOf([*c]u8))));
```

However, moving the compilation to a dedicated C file works fine:
```
#define FOO_IMPLEMENTATION
#include <foo.h>
```

The behavior is different between these two approaches:
- importing the single header with the implementation flag via `cImport()` will trigger an alignment error
- compiling the code with `addCSourceFile()` works, then we can import the header file without the implementation and access the C library

I am not sure we are supposed to compile a library directly from `cImport()`, but I thought it was worth asking about it and point out the difference.

**Comment by ehaas** [comment]
Same issue as https://github.com/ziglang/zig/issues/10222 (the error message in that one is deprecated since `anyopaque` replaced `c_void`)

**Comment by sskras** [comment]
I confirm it's fixed at least in `0.11.0-dev.4002+7dd1cf26f`:
```
$ make
total 12
-rw-r--r-- 1 sskras sskras 196 Jul 16 08:35 GNUmakefile
-rw-rw-r-- 1 sskras sskras 140 Jul 16 07:50 foo.h
-rw-rw-r-- 1 sskras sskras 239 Jul 16 07:50 import.zig

zig version
0.11.0-dev.4002+7dd1cf26f

zig build-lib -I. -lc import.zig

total 2052
-rw-r--r-- 1 sskras sskras     196 Jul 16 08:35 GNUmakefile
-rw-rw-r-- 1 sskras sskras     140 Jul 16 07:50 foo.h
-rw-rw-r-- 1 sskras sskras     239 Jul 16 07:50 import.zig
-rw-rw-r-- 1 sskras sskras 1042524 Jul 16 08:35 libimport.a
-rw-rw-r-- 1 sskras sskras 1042296 Jul 16 08:35 libimport.a.o

ar t libimport.a
./libimport.a.o

objdump -x libimport.a | grep -e foo -e main
0000000000000980 l     F .text  0000000000000048 import.main
0000000000000000 g     F .text  0000000000000590 main
0000000000001840 g     F .text  000000000000004e foo
```
